Pkg 4023: v12.3 - initial

danpetry opened the pull request:
**Destination channel:** defaults

### Links

- [PKG-4583](https://anaconda.atlassian.net/browse/PKG-4583) 

### Explanation of changes:

- Part of CUDA 12.4 build, following [conda-forge's pattern](https://github.com/conda-forge/staged-recipes/issues/21382) which was developed in conjunction with Nvidia. I don't see a need to stage these then release them all at once. They won't cause a problem released one at a time.
- Please check the main branch too, this has just been forked
- Changes made to adjust feedstocks for differences between conda-forge and defaults. Please see commit messages for description of changes.


[PKG-4583]: https://anaconda.atlassian.net/browse/PKG-4583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_feg7a3zf8n/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_99wgzstjc3/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_6fl884wpq4/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_73jdk5s4mf/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:345: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:345: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:290: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:345: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_93fbzygcgd/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:345: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:345: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:345: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:290: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_9d0w662p1q/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_592w1ocwvz/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_201vu81wk8/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:345: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:345: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:345: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:290: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_cev6qcbw4x/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_f4dwcr_3jc/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_55j8_62r4b/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_0a0l_1dos0/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:344: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:289: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:344: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:344: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_a3c_elz3vj/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:349: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:294: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:349: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:349: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_0dyjf7efqa/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:351: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:296: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:351: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:351: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_earl8o_xkw/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:349: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:349: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:349: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:294: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_b8de2z0ht5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:349: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:349: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:349: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:294: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_a1d6dampwu/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:349: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:349: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:349: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:294: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_09x2q80dj2/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:349: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:294: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:349: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:349: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

danpetry committed:
Correct windows patch line endings and adjust .gitattributes to match

danpetry committed:
Unpin the gcc version

It looks like our aggregate-pinned version, 11, is fine too:
https://docs.nvidia.com/cuda/archive/12.3.0/cuda-installation-guide-linux/index.html#host-compiler-support-policy

danpetry committed:
Change visual studio compiler version to the one we've got in defaults

danpetry committed:
Adjust supported glibc version on linux-aarch64; it's different from conda-forge's

danpetry committed:
Skip s390x, CUDA doesn't support it

danpetry committed:
Remove conda-forge build requirement on specific arm architecture

We don't have arm-variant as a package in defaults.

It's left in run_constrained to support compatibility when mixed with
conda-forge packages.

danpetry committed:
Symlink gcc, g++ and ar so CMake and nvcc find them

The gcc and g++ symlinks are provided on conda-forge by their gcc and ar
packages.

linux-aarch64 isn't built on conda-forge, and the ar symlink is added to make
the tests pass on that platform.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_f462mw6k0l/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

SC426 reviewed (approved):
LGTM!

cbouss reviewed (approved):
From what I see, changes are in line with what was done in other cuda- feedstocks.

danpetry commented:
Going to leave this open until we get an agreement signed. Managing the upload channel is a pain.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_a9els_1yyr/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_336g2hsnkw/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

danpetry committed:
Upload all noarch packages

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_15cha6_pno/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_fc2moenvjf/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_17iaksotn2/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:352: license_file_overspecified: Using license_file and license_url is overspecified.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-aarch64": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:96: host_section_needs_exact_pinnings: output "cuda-nvcc-dev_linux-ppc64le": Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:140: host_section_needs_exact_pinnings: output "cuda-nvcc-impl": Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:352: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:58: missing_tests: output "cuda-nvcc-tools": No tests were found.
- clone/recipe/meta.yaml:352: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:1: patch_must_be_in_build: patch must be in build when source/patches is set.
- clone/recipe/meta.yaml:297: missing_tests: output "cuda-nvvm-dev_win-64": No tests were found.
===== Final Report: =====
5 Errors and 5 Warnings were found

</details>
